### PR TITLE
tests: gatt_dm: Migrate to the new Ztest API

### PR DESCRIPTION
**Pull request by KAGA164**
Migrate the GATT Discovery Manager tests to the new Ztest API.

### CODE REVIEW

**Review by dchat-nordic — APPROVED** [review]

File: tests/subsys/bluetooth/gatt_dm/src/main.c [hunk 1]
```diff
@@ -81,20 +81,19 @@ void test_cb_error_found(struct bt_conn *conn, int err, void *context)
...
 
-void test_setup(void)
+void test_setup(void *fixture)
 {
+	ARG_UNUSED(fixture);
+
 	k_sem_reset(&discovery_finished);
 	bt_gatt_discover_mock_setup(discover_sim, ARRAY_SIZE(discover_sim));
 }
-
```
[the remark this answers is not in the document]
Comment (KAGA164) [reply tests/subsys/bluetooth/gatt_dm/src/main.c]: Added

File: tests/subsys/bluetooth/gatt_dm/src/main.c [hunk 2]
```diff
@@ -81,20 +81,19 @@ void test_cb_error_found(struct bt_conn *conn, int err, void *context)
...
 	.completed         = test_cb_completed,
 	.service_not_found = test_cb_service_not_found,
 	.error_found       = test_cb_error_found
 };
 
-void test_setup(void)
+void test_setup(void *fixture)
 {
+	ARG_UNUSED(fixture);
+
```
[the remark this answers is not in the document]
Comment (KAGA164) [reply tests/subsys/bluetooth/gatt_dm/src/main.c]: It is actually test_before function, I named it with confusing name. Changed test_setup->test_before

File: tests/subsys/bluetooth/gatt_dm/src/main.c [hunk 3]
```diff
@@ -467,29 +469,10 @@ void test_gatt_many_serv_by_uuid(void)
...
-		ztest_unit_test_setup_teardown(test_gatt_many_serv_by_uuid, test_setup,
-					       unit_test_noop)
-	);
-
-	ztest_run_test_suite(test_gatt);
+	/* ------------------------------------------------------ */
+	/* Clean up */
+	bt_gatt_dm_data_release(dm);
+	zassert_equal(0, bt_gatt_dm_attr_cnt(dm), "Parameter count after clearing: %d",
+		      bt_gatt_dm_attr_cnt(dm));
```
Comment (maje-emb) [inline tests/subsys/bluetooth/gatt_dm/src/main.c:477]: Is it possible to move the clean-up to the `test_after` function?

Comment (KAGA164) [reply tests/subsys/bluetooth/gatt_dm/src/main.c:477]: No really. We have at least one test case that no required clearing and return NULL pointer to the dm

**Review by maje-emb — APPROVED** [review]

**Review by gopiotr — APPROVED** [review]
I tested it locally on my hardware and everything work as expected. Thank you for your work.

**Pull request merged**